src/Label/LabelOptionsResolver.php

```diff
@@ -68,23 +69,72 @@ public function getLabelOptions(Order $order): array
     public function getDeliveryOptions(Order $order): array
     {
         $deliveryOptions = OrderSettingsFactory::create($order)->getDeliveryOptions();
+        $packageType     = $this->getPackageType($order, $deliveryOptions);
         $shipmentOptions = $this->getShipmentOptions(
             $deliveryOptions,
             $order->getProducts(),
             $order->getIdCarrier()
         );
 
-        if (isset($shipmentOptions['insurance'])) {
-            $shipmentOptions['insurance'] = false === $shipmentOptions['insurance'] ? 0 : 500;
+        if (isset($shipmentOptions['insurance']) && $deliveryOptions) {
+            $shipmentOptions['insurance'] = true === $shipmentOptions['insurance']
```

Comment:
Is die true niet overbodig?

Reply:
Wanneer insurance in eerste instantie gezet is via products, is deze true of false, terwijl de rest van de flow alleen een getal accepteert (0 voor 'geen'). Dit is die conversie / check, daarom wordt er explicite op true gecheckt.